eleven/one_nieoh.py

```diff
@@ -0,0 +1,13 @@
+def first_of(arr, x):
+    def func(L, U):
+        M = (L+U)//2
+        if arr[M] == x:
+            if L == M: 
```

Comment:
You can put this `if L==M: return M` bit at the top level as the first check.

Reply:
your comment is wrong

Reply:
Sorry! I meant 
```python
if arr[L] == x:
  return L
```